### DISCUSSION TITLE
Conditional webhook

### DISCUSSION THREAD

**Discussion by jk-onnet** [netbox-community/netbox#9023] [Q&A] [answered]
When netbox object is changed (e.g. virtual machine name updated), how to know the specific field (name) is change from netbox side to trigger webhook? Will conditional webhook works?

My main goal is to change other system name using NetBox webhook and ansible.

I wish to trigger webhook on name change but not triggered by other attribute like description, custom fields and etc to reduce noise.

**Comment by candlerb** [comment, marked as answer]
> how to know the specific field (name) is change from netbox side

The webhook payload contains both `"pre"` and `"post"` sections. Your webhook receiver can compare these to decide whether action is required or not.

**Reply by jk-onnet** [reply]
Thank you, I'm able to do that on webhook receiver. May I know is it possible to compare on webhook sender based on your understanding?

**Reply by candlerb** [reply]
I don't think so.  There are [conditional webhooks](https://netbox.readthedocs.io/en/stable/additional-features/webhooks/#conditional-webhooks) but as far as I know you can only filter on the current value of an object, not the previous.
